### PR TITLE
Msw 92742430 click data revised

### PR DESCRIPTION
**Pull request by michaelwehrley**
Record Click Data

* Record Click Events
* Rename Path Methods
* Update Tests
* Integrate localStorage
* Add Client Side Authentication

### CODE REVIEW

File: src/js/Constants.js [hunk 1]
```diff
@@ -2,6 +2,8 @@
 (function() {
   'use strict';
 
+  var BASE_URL = 'http://localhost:3000';
```
Comment (ericcf) [inline src/js/Constants.js]: Did you mean to commit with `localhost`?

Comment (michaelwehrley) [inline src/js/Constants.js]: Yes and no.  I guess your right, this merge request should be staging and I can change it locally.

File: src/js/controllers/StartController.js [hunk 2]
```diff
@@ -2,21 +2,90 @@
...
+    this.token = {
+      configurationToken: ''
+    };
+
+    this.uploadUserClickData = function() {
+      var allClickData, params, Things;
+
+      allClickData = resourceCache.fetchAll('ednavigatorClickEvents');
+
+      Things = $resource(
```
Comment (ericcf) [inline src/js/controllers/StartController.js]: Might want to consider renaming `Things` to something more meaningful :-)

File: src/js/services/Guid.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+/* Saves datum to and fetches datum from localStorage */
+(function() {
+  'use strict';
+
+  function Guid() {
+    this.new = function() {
```
Comment (ericcf) [inline src/js/services/Guid.js]: The output of this looks like a UUID, but doesn't quite fit the format: http://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_.28random.29

**Comment by ericcf** [comment]
None of these are dealbreakers. Merging.

**Pull request merged**